### PR TITLE
feat: add web executor + add testcase context

### PR DESCRIPTION
**Pull request by sguiheux**

### CODE REVIEW

File: executors/web/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+* result.timehuman
+* result.title
+* result.error
+
+## Action
+* navigate: navigate to url
+* title: get title value
+* screenshot: take a screenshot
+
+more actions are coming...
```
Comment (yesnault) [inline executors/web/README.md]: user (or us) will create issue for that, no ? (so, no need to tease here)

File: executors/web/web.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	Screenshot string `json:"screenshot" yaml:"screenshot"`
+}
+
+// Result represents a step result
+type Result struct {
+	Executor    Executor `json:"executor,omitempty" yaml:"executor,omitempty"`
+	TimeSeconds float64  `json:"timeseconds,omitempty" yaml:"timeseconds,omitempty"`
+	TimeHuman   string   `json:"timehuman,omitempty" yaml:"timehuman,omitempty"`
+	Title       string   `json:"title,omitempty" yaml:"title,omitempty"`
+	Err         error    `json:"error,omitempty" yaml:"error,omitempty"`
```
Comment (yesnault) [inline executors/web/web.go]: err have should be a string, as other executor

File: executors/web/web.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+		}
+	case "title":
+		title, err := page.Title()
+		if err != nil {
+			return nil, err
+		} else {
+			r.Title = title
+		}
+	}
+
```
Comment (yesnault) [inline executors/web/web.go]: page content is not available ?

File: process.go [hunk 4]
```diff
@@ -231,8 +231,16 @@ func runTestSuite(ts *TestSuite, detailsLevel string) {
 }
 
 func runTestCase(ts *TestSuite, tc *TestCase, l *log.Entry, detailsLevel string) {
+	l.Infof("Init context")
```
Comment (yesnault) [inline process.go]: Debug

**Comment by yesnault** [comment]
Can you add something on global Readme, section "write your executor" on what is context arg for Run(...) with example of web exec ?

File: executors/web/web.go [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+const Name = "web"
+
+// New returns a new Executor
+func New() venom.Executor {
+	return &Executor{}
+}
+
+// Executor struct
+type Executor struct {
+	URL        string `json:"url" yaml:"url"`
```
Comment (yesnault) [inline executors/web/web.go]: ,omitempty everywhere (to avoid noise in failures)

File: executors/web/web.go (hunk 3 above)
Comment (sguiheux) [inline executors/web/web.go]: page content is available, it will come in the future

File: process.go (hunk 4 above)
Comment (sguiheux) [inline process.go]: I put info like logs below :/
l.Infof("start")
l.Infof("end")

Debug for all ?

Comment (yesnault) [inline process.go]: yep, for context. Below is for run.

**Comment by sguiheux** [comment]
@yesnault Need to write a section 'write your own context' in the global readme too

File: README.md [hunk 6]
```diff
@@ -242,14 +246,21 @@ func (Executor) GetDefaultAssertions() venom.StepAssertions {
...
-func (Executor) Run(l *log.Entry, aliases venom.Aliases, step venom.TestStep) (venom.ExecutorResult, error) {
+func (Executor) Run(ctx context.Context, l *log.Entry, aliases venom.Aliases, step venom.TestStep) (venom.ExecutorResult, error) {
 
 	// transform step to Executor Instance
 	var t Executor
 	if err := mapstructure.Decode(step, &t); err != nil {
 		return nil, err
 	}
 
+	// Get testcase context
```
Comment (yesnault) [inline README.md]: // Get testcase context if needed

File: README.md [hunk 7]
```diff
@@ -272,6 +283,48 @@ func (Executor) Run(l *log.Entry, aliases venom.Aliases, step venom.TestStep) (v
 Feel free to open a Pull Request with your executors.
 
 
+## TestCase Context
+
+TestCase Context allows you to inject datas in all Steps
```
Comment (yesnault) [inline README.md]: Add: "define a context is optional, but can be usefull to keep data between teststeps on a testcase."
?

File: context/web/context.go [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+const ContextPageKey = "page"
+
+// New returns a new TestCaseContext
+func New() venom.TestCaseContext {
+	return &TestCaseContext{}
+}
+
+// TestCaseContex represents the context of a testcase
+type TestCaseContext struct {
+}
```
Comment (fsamin) [inline context/web/context.go]: one line

**Pull request merged**